### PR TITLE
[SwiftASTContext] Share common code with PlatformDarwin.

### PR DESCRIPTION
**Pull request by JDevlieghere**
Commit 7aa28995e87f6cec2e02608c978885cf01dcc65d refactored the Xcode
resource directory logic and made it possible to have this code shared
with SwiftASTContext.

### CODE REVIEW

**Comment by JDevlieghere** [comment]
@swift-ci please test

**Comment by JDevlieghere** [comment]
I'll cherry-pick this to `master-next` once this is approved and landed.

File: lldb/source/Symbol/SwiftASTContext.cpp [hunk 1]
```diff
@@ -1536,14 +1536,18 @@ static StringRef GetXcodeContentsPath() {
...
-
         size_t pos = output.rfind(substr);
         if (pos != std::string::npos) {
+          // Remove everything after the substring.
           output.erase(pos + strlen(substr));
+
+          // Strip any leading newline characters and everything before them.
+          // This needs to happen *after* the erase above because otherwise a
+          // trailing newline might erase the whole string.
+          size_t last_newline = output.find_last_of("\r\n");
```
Comment (adrian-prantl) [inline lldb/source/Symbol/SwiftASTContext.cpp]: I don't see why the `\r` is necessary here. We are not going to run this on macOS < 10.0. On Windows `\r\n` will still contain `\n`.

Comment (adrian-prantl) [reply lldb/source/Symbol/SwiftASTContext.cpp]: Shouldn't/Isn't this function in lldb/source/Plugins/Platform/MacOSX/PlatformMacOSX.cpp?

Comment (adrian-prantl) [reply lldb/source/Symbol/SwiftASTContext.cpp]: ConstString PlatformMacOSX::GetSDKDirectory(lldb_private::Target &target) {

Comment (JDevlieghere) [reply lldb/source/Symbol/SwiftASTContext.cpp]: Good point, this function is doing something different but it's also calling `xcrun`. Let me factor that out and reuse it in both places.

**Review by dcci — COMMENTED** [review]
This is really interesting, thanks. Is this commit motivated by a bug you hit while parsing the SDK or it's just a drive-by fix?

**Comment by JDevlieghere** [comment]
> This is really interesting, thanks. Is this commit motivated by a bug you hit while parsing the SDK or it's just a drive-by fix?

Yup, it was what caused my REPL tests to fail. Apparently `fish` is printing some warning before the path now that we honor the `$SHELL` variable. That also explains why they still failed when I ran the test under a different shell.

**Review by dcci — COMMENTED** [review]
LGTM, but Adrian should sign-off too.

**Comment by dcci** [comment]
@swift-ci test

**Comment by JDevlieghere** [comment]
This depends on an upstream change that hasn't landed yet. I think we can remove even more complexity so I'm going to rework the upstream patch a bit.

**Comment by JDevlieghere** [comment]
Depends on #921

**Comment by JDevlieghere** [comment]
@swift-ci please test

**Review by adrian-prantl — APPROVED** [review]
This is clearly an improvement, but really these functions should all be in PlatfromDarwin on llvm.org.

**Comment by JDevlieghere** [comment]
The AM claimed that there was nothing to merge from 20200108 into swift/master but the Linux bot says the dependency isn't there...

**Comment by JDevlieghere** [comment]
@swift-ci please test

**Comment by JDevlieghere** [comment]
@swift-ci please test

**Pull request merged**